### ISSUE TITLE
`createServerData$` request is undefined when invoked from the client side

### ISSUE THREAD

**Issue by ghalle** [solidjs/solid-start#289]
Originally reported by tomchristensen on discord.

> How does createServerData$ within routeData work in the context of client-side navigation? It seems like the request object is only passed to the fetcher (in createServerData$) on the initial page request, but subsequent client-side navigation to the same page runs the fetcher with undefined for the request object
Example: https://stackblitz.com/edit/github-fk8exj?file=src%2Froutes%2Findex.tsx - on loading the index page, the console logs the request object, but if you navigate to "About" and then back to "Index", the console logs undefined for the request

**Comment by nksaraf** [comment]
As per discussion with @ghalle, I think this is not a code bug but more a `vite` config bug from the recent revert of fixes we had made in [solidjs/vite-plugin-solid](https://github.com/solidjs/vite-plugin-solid). Our tests do test this in prod, but in dev we are at the mercy of `vite` dependency optimization behaviour.

Essentially we end up getting two copies of `ServerContext.tsx` on the client (one with query hash, one without). So on navigation, we don't send a marker for the `FetchEvent` with the server function call. This makes it so that we don't get access to the `event` as the second argument to these calls. 

Our initial fix is going to be to disable dependency optimization, until we can add esbuild to actually bundle our dependencies as optimization.

```ts
import solid from 'solid-start/vite';
import { defineConfig } from 'vite';

export default defineConfig({
  plugins: [solid()],
  optimizeDeps: {
    extensions: ['jsx', 'tsx'],
    disabled: true,
  },
});
```

We will makes these changes in solid-start's config itself so users don't need to worry about it.

**Comment by nksaraf** [comment]
Fixed by e94ea868216b5f475ca5f9e7034fbe1d9fbb6f64
